nil-docs/zkllvm/overview/key-components-tools.md

```diff
@@ -0,0 +1,83 @@
+# Key components and tools
+
+## zkLLVM Components
+
+The architecture of zkLLVM is modular and extendable. It can be divided into three main components.
+
+### Frontends
+
+Frontends parse high-level languages and generate abstract syntax trees (ASTs). Presently, zkLLVM supports C++ and Rust as frontends based on `clang` and `rustc`, respectively.
+
+:::info[zkLLVM and LLVM]
+
+In comparison to the vanilla LLVM project frontends, zkLLVM frontends generate ASTs that are compatible with the zkSNARK circuit generation process. These ASTs include additional types and metadata.
+
+:::
+
+### Circifier
+
+`circifier` receives an AST and converts it into an intermediate representation (IR) of a circuit while applying circuit-specific optimizations.
```

Comment:
Circifier repository is a fork of LLVM with some extensions in the frontend and IR. So I think it's incorrect to mark it as a component of our pipeline. The main reason for it is that the circifier *contains* C++ frontend (clang). And again, AST is not an input for circifier, it's only an internal data structure in frontends.

Reply:
got it, thank you!

Reply:
I see, resolved by removing mentions of `circifier` here